src/command/restore/restore.c

```diff
@@ -1320,6 +1320,21 @@ restoreSelectiveExpression(Manifest *manifest)
             // Log databases found
             LOG_DETAIL_FMT("databases found for selective restore (%s)", strZ(strLstJoin(dbList, ", ")));
 
+            // Get all system db ids, even in case users would have manually re-created it
```

Comment:
I made some changes and I hope you will agree with them. I realized a few things in doing live tests:

1) Unlike the unit tests, the system databases are displayed in the `databases found for selective restore (%s)` message. So the unit tests were a bit lacking here because the PG_VERSION is not always added in the manifest in the unit tests when, realistically, they should be. But since you fixed the issue, we'd always display the system ids, as we should, in the `databases found for selective restore (%s)` message. I only updated one unit test as an example, because I wanted to give you an opportunity to review the changes I made and incorporate them into your branch if you agree and maybe update the unit tests yourself.

2) We have a cool anti-merge function (strLstMergeAnti) that was useful here so, using that, we are able to use the dbList array as the main array for verifying and checking databases so I moved the generation of the systemDbIdList up and added the ids to the dbList (rather than remove). And also was able to remove the need to keep checking the PG_USER_OBJECT_MIN_ID.

3) We have gotten complaints on the amount of logging we do, so, given the strLstMergeAnti() function we're able filter the dbList with it and then provide the list of zeroed databases in a concatenated string, just like we do with the `databases found for selective restore (%s)` message.

I tested before and after the change on a live system and everything is working as expected. I have not updated the unit tests or the mock tests. I can definitely take care of the mock tests, but, again, I wanted to see what you thought of the changes and see if you wanted to try to merge them into your branch and work on the unit tests. 

The comparison of my branch to your branch is at:
https://github.com/pgstef/pgbackrest/compare/issues/986...cmwshang:pgstef-bugfix-only-issue986

And my branch itself is at: https://github.com/cmwshang/pgbackrest/tree/pgstef-bugfix-only-issue986)

Please let me know your thoughts.

Reply:
I like your changes **a lot**, thanks @cmwshang!
I struggled to merge your branch but I managed to extract and import your patch in this PR. I hope that's not a problem :-(
I fixed the unit tests too. I'm now waiting to see if the MockAllTest will succeed.